modules/api/src/main/java/org/apache/ignite/lang/ErrorGroups.java

```diff
@@ -577,4 +577,17 @@ public static class PlacementDriver {
         /** Primary replica await error. */
         public static final int PRIMARY_REPLICA_AWAIT_ERR = PLACEMENT_DRIVER_ERR_GROUP.registerErrorCode((short) 2);
     }
+
+    /** Criteria error group. */
+    @ErrorCodeGroup
+    public static class Criteria {
+        /** Criteria error group. */
+        public static final ErrorGroup CRITERIA_ERR_GROUP = registerGroup("CRITERIA", (short) 19);
+
+        /** No more pages in the cursor error. */
+        public static final int CRITERIA_CURSOR_NO_MORE_PAGES_ERR = CRITERIA_ERR_GROUP.registerErrorCode((short) 1);
+
+        /** Cursor is already closed error. */
+        public static final int CRITERIA_CURSOR_CLOSED_ERR = CRITERIA_ERR_GROUP.registerErrorCode((short) 2);
```

Comment:
Can this error be in common group?

Reply:
I think these codes (+ similar SQL codes) are not specific to Criteria (or SQL) modules. Moreover, Cursors now are part of Public API, and we can introduce a common exception e.g. CursorAccessException for that purpose, and use it like an IllegalArgumentException for arguments validation.

Reply:
Moved to common group. Moved `CursorAccessException` to lang package. Added `NoMorePagesException`